heat/core/random.py

```diff
@@ -30,7 +30,7 @@ def randn(*args, split=None, comm=MPI_WORLD):
 
     Parameters
     ----------
-    d0, d1, …, dn : int, optional
+    d0, d1, …, dn : ints, optional
```

Comment:
Implementation of the function is closely related to issue #54. What is the intended behaviour of this function? Output a number of random numbers where each of the split parts is random and not identical to the other split chunks of the other nodes? What will happen if I use the very same seed but a different node count? Will the random tensor differ?

Reply:
It looks like torch's randn will not generate the same values for arrays of different sizes. This means that even though the same seed is used, the split tensor will not be the same as a tensor generated only on one process. i.e.
```python
torch.randn((3, 3, 3))[0] != torch.randn((1, 3, 3))
```
One solution to this would be to generate the whole dataset then split it. But this will not scale. I do not see another way to do this. 

It must be noted that the matrix generated by the ht.randn with a fixed seed (torch.manual_seed) and a fixed size produces a reproducible result

Reply:
if the sizes of the data are the same on multiple nodes, then the random data will be the same.

Reply:
In my opinion, we can leave it as proposed for now, however, for the future, I would want to actually have a different behaviour. Consider the following example:

```
ht.set_seed(1)
ht.randn(100, 5, 3, split=0)
```

Should always produce the same set of random numbers independent of the utilized nodes (for reproducibility reasons as you mentioned). This is exactly what is requested in issue #54. Obviously, this means that we would have to come up with a pseudo random generator allowing to skip to arbitrary/some fixed positions into the random sequence. 

In the proposed fix, we would have a simplified randn() call. It will only provide reproducible results for the exact same node count.

Reply:
100% agree. i am trying to figure a way to do this. i will move this discussion to the issue and close this request.